### PR TITLE
Add Streamlit dashboard example

### PR DESCRIPTION
**Pull request by dpol1**
## Summary                                                                                                                                                            
   
  Local Streamlit dashboard for interactive exploration of SPRUCE-enriched Parquet output. Same GreenOps/FinOps metrics as the static report (cost, energy, operational 
  + embodied CO2e, water, trends, top emitters, regional/tag splits) but driven by DuckDB so it accepts files, globs, directories, or `s3://` URIs.
                                                                                                                                                                        
  Optional add-on. Does not change existing CLI, processors, or report generation. Dependencies isolated in `requirements-dashboard.txt`.                               
   
  ## Changes                                                                                                                                                            
                  
  - `dashboard.py` — Streamlit app, DuckDB-backed queries, `--input` flag.
  - `requirements-dashboard.txt` — dashboard-only deps (streamlit, duckdb, pandas, plotly).
  - `.streamlit/config.toml` — project-local theming.                                                                                                                   
  - `scripts/gen_synthetic.py` — synthetic CUR-shaped Parquet generator for demos / local review.
  - `docs/src/howto/dashboard.md` + `SUMMARY.md` + `tutorial/results.md` + `README.md` — setup and usage.                                                               
  - `.gitignore` — ignore Python caches.                                                                                                                                
                                                                                                                                                                        
  ## Run                                                                                                                                                                
                  
  ```bash                                                                                                                                                               
  pip install -r requirements-dashboard.txt
  python scripts/gen_synthetic.py -o output/synth.parquet
  streamlit run dashboard.py -- --input output/
```
                                                                                                                                                                        
  Also accepts a single Parquet file, a glob, or s3://bucket/prefix/.                                                                                                   

  Validation

  - Synthetic dataset generated and loaded via DuckDB.
  - Streamlit app starts; all panels (cost, energy, emissions split, water, trends, top-N, region, tags) render.
  - Static report and existing pipelines untouched.

 Implement #72

### CODE REVIEW

**Comment by dpol1** [comment]
### TODO **Final design by Claude Design:**

Screenshot 2026-05-01 at 16-58-21 Spruce Design System

**Comment by jnioche** [comment]
image
the trends are hard to read as  all the impacts as stacked - they are not comparable with each other
Maybe one graph per metric or have each metric as a line instead so that their own trend is easier to read?
since they ahve different units I would go for the former

**Comment by jnioche** [comment]
image

top emitters - not sure the graph adds anything

**Comment by jnioche** [comment]
image

Regions, not much value stacking energy and emissions in the graph. A pie chart with just the emissions would be good

**Comment by jnioche** [comment]
Tags - same as top emitters. at least sort the graph so that the entries are not shown alphabetically but as ordered in the table?

**Comment by jnioche** [comment]
Final comment, instead of specifying all the values by default for the filters billing period and region, can they be empty by default? This would be easier to read and possibly much faster

**Comment by jnioche** [comment]
The root of the repo is getting quite messy, I wonder whether we should have the dashboard and reporting scripts in a directory e.g. /reporting ?
what do you think @dpol1 

Fantastic job on the dashboard BTW! I love it!

**Comment by dpol1** [comment]
> The root of the repo is getting quite messy, I wonder whether we should have the dashboard and reporting scripts in a directory e.g. /reporting ? what do you think @dpol1

Yes that's a great idea
> 
> Fantastic job on the dashboard BTW! I love it!

Thanks for all the feedback, I've just made the scaffolding now I have to refine

**Comment by jnioche** [comment]
For the trends with one graph per metric: maybe aggregate the emissions to make it simpler and use different colours e.g. yellow for energy, brown for emissions, blue for water?

**Comment by dpol1** [comment]
> Final comment, instead of specifying all the values by default for the filters billing period and region, can they be empty by default? This would be easier to read and possibly much faster



> For the trends with one graph per metric: maybe aggregate the emissions to make it simpler and use different colours e.g. yellow for energy, brown for emissions, blue for water?

I'll properly handle all of it

**Comment by jnioche** [comment]
please let me know when you want me to review the PR @dpol1

**Comment by jnioche** [comment]
should `.streamlit ` be moved  to reporting? and perhaps move some of the additions in _.gitignore_ to a `/reporting/.gitignore` ?

**Comment by dpol1** [comment]
> please let me know when you want me to review the PR @dpol1 

Sure, while the PR is in draft flag is not ready for review

**Comment by jnioche** [comment]
> > please let me know when you want me to review the PR @dpol1
> 
> Sure, while the PR is in draft flag is not ready for review

I think you had left the draft flag on a previous PR - hence the precision.

**Comment by dpol1** [comment]
Don t worry - hopefully tomorrow everything will be ready

**Comment by dpol1** [comment]
@jnioche Good to go for review!

### Highlights since last pass 

- header alignment + sizing fixed, no more clipping
- regions pie legend was truncating; moved it to bottom-center horizontal, bumped height, labels visible
- added `menu_items` in `st.set_page_config` (help → docs site, bug → issues, about → version blurb + repo/docs/discussions)
- moved `.streamlit/` under `reporting/`. run command in `docs/src/howto/dashboard.md` is now `cd reporting && streamlit run dashboard.py -- --input ../output/` so Streamlit picks up the project-local config
- `scripts/gen_synthetic.py` now spans multiple regions/services/billing periods, so the dashboard has actual shape to render against
- small cleanups in tables and SQL helpers
- `dashboard.md`, `tutorial/results.md` and the README pointer trimmed to match the rest of SPRUCE's docs; dropped the brand-color/customization bits

### Open questions:

- tried moving the python block to `reporting/.gitignore` — `report.py`, `cur_sanitize.py` and the `scripts/` entry points run from root, so their caches show up in `git status` (a nested gitignore only matches paths under itself). leaving it at root for now. can revisit once python tooling is consolidated under `reporting/`.
- screenshots in `dashboard.md`: holding off till the layout settles. add a short gallery once it's final, or skip and let the live site / repo screenshots do the work — preference?

**Comment by jnioche** [comment]
@dpol1 this is amazing, thanks!

I see that the regions filter has all the possible regions available, which is potentially confusing and not needed. Can we get the values from what is found in the dataset instead? Make sure you use the 'region' column, which we create in SPRUCE. This way the dashboard will work in the future when we cover other CSPs, at least for this filter.

> gitignore - can revisit once python tooling is consolidated under reporting?

absolutely

> screenshots in dashboard.md

I think the layout is fine now. Could we have a couple of screenshots as part of this PR?
I don't think we'll do a live version at least in the short term

Thoughts for the future: would be good to have on the overview a few of the typical comparisons:
-  co2 equivalent in km driving a family car 
-  co2 equivalent of flying from X - Y (choose a typical one)
- energy use of a typical household for a year
- energy use - number of cups of tea made (can't help it I live in the UK)
- water: number of olympic size swimming pools / bath tubes

Can take inspiration from 

https://demo.cloudcarbonfootprint.org/
https://huggingface.co/spaces/genai-impact/ecologits-calculator

**Comment by dpol1** [comment]
> I see that the regions filter has all the possible regions available, which is potentially confusing and not needed. Can we get the values from what is found in the dataset instead? Make sure you use the 'region' column, which we create in SPRUCE. This way the dashboard will work in the future when we cover other CSPs, at least for this filter.

Great catch! It's essential for the dashboard

>I think the layout is fine now. Could we have a couple of screenshots as part of this PR?

Yes I'll add some screenshots
> 
> Thoughts for the future: would be good to have on the overview a few of the typical comparisons:

>    * energy use - number of cups of tea made (can't help it I live in the UK)

Why not compare it to energy use for plate of pasta or a pizza :-)
Joking aside, fair points but not the purpose of this PR, for now make a follow-up issue is the ideal and refine later

**Comment by jnioche** [comment]
> Why not compare it to energy use for plate of pasta or a pizza :-)

Could have another selector on the left "Choose your stereotype for equivalences" ;-)

and yes, definitely an addition for later

**Comment by jnioche** [comment]
great job, thanks @dpol1 
we can refine things if needed after the 1.0 release
